### PR TITLE
APIS-5813 Add ability for DevHub user to request change to production app name

### PR DESCRIPTION
**Pull request by petekirby-ee**
This PR is for both APIS-5813 (Create Deskpro ticket for SDST when Dev hub user has requested application name change) and APIS-5744 (Allow dev hub user to initiate "change of application name")

### CODE REVIEW

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/controllers/Details.scala [hunk 1]
```diff
@@ -205,4 +208,51 @@ class Details @Inject() (
...
+  }
+
+  def requestChangeOfAppNameAction(applicationId: ApplicationId): Action[AnyContent] = canChangeProductionDetailsAndIsApprovedAction(applicationId) { implicit request =>
+      val application = request.application
+
+      def handleValidForm(form: ChangeOfApplicationNameForm): Future[Result] = {
+        val requestForm = ChangeOfApplicationNameForm.form.bindFromRequest
+        val newApplicationName = form.applicationName
+
+        if (newApplicationName.equalsIgnoreCase(application.name)) {
```
Comment (robdawson-hmrc) [inline app/uk/gov/hmrc/thirdpartydeveloperfrontend/controllers/Details.scala:226]: We could also trim whitespace from the names before comparing them?

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/service/ApplicationService.scala [hunk 2]
```diff
@@ -207,6 +207,21 @@ class ApplicationService @Inject() (
...
+    def createDeskproTicket(application: Application, newApplicationName: String, requesterName: String, requesterEmail: String) = {
+      val previousAppName = application.name
+      val appId = application.id
+
+      DeskproTicket.createForRequestChangeOfProductionApplicationName(requesterName, requesterEmail, previousAppName, newApplicationName, appId)
+    }
+
+    val ticket = createDeskproTicket(application, newApplicationName, requesterName, requesterEmail)
+    for {
+      ticketResponse <- deskproConnector.createTicket(ticket)
```
Comment (robdawson-hmrc) [inline app/uk/gov/hmrc/thirdpartydeveloperfrontend/service/ApplicationService.scala:221]: is the for...yield needed here, could we just return deskproConnector.createTicket(ticket) ?

**Review by AndySpaven — APPROVED** [review]

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/controllers/Details.scala (hunk 1 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/thirdpartydeveloperfrontend/controllers/Details.scala:226]: Good point - fixed

File: app/uk/gov/hmrc/thirdpartydeveloperfrontend/service/ApplicationService.scala (hunk 2 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/thirdpartydeveloperfrontend/service/ApplicationService.scala]: Fixed

**Review by robdawson-hmrc — APPROVED** [review]

**Pull request merged**